File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-0.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        l = 0
        r = len(nums)- 1
        res = 0
        while l < r :
            mid = (l + r ) // 2
            if nums[mid] < nums[r]:
                r = mid
            else:
                l = mid + 1
            # if nums[mid] > nums[l]:
            #     if nums[mid] > nums[r]:
            #         l = mid + 1
            #     else:
            #         r = mid - 1
            #         res = min(nums[res],nums[mid])
            # else:
            #     r = mid - 1
            #     res = min(nums[res],nums[mid])
        return nums[r]

File: Data Structures & Algorithms/max-water-container/submission-0.go
func maxArea(heights []int) int {
	res := 0
	l := 0 
	r := len(heights) - 1

	for l < r {
		
		p := (r - l) * min(heights[l],heights[r])
		if p > res {
			res = p
		}

		if heights[l] < heights[r] {
			l++
		}else{
			r--
		}

	}


	return res
}
